.github/workflows/check-labels.yml

```diff
@@ -0,0 +1,52 @@
+name: Check Labels
+
+on:
+  # We need pull_request_target to be able to post comments on PRs from forks.
+  # Only allow pull_request_target when merging to main, not some historical branch.
+  #
+  # Make sure to don't introduce explicit checking out and installing/running
+  # untrusted user code into this workflow!
+  pull_request_target:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches: [main]
+
+  # To check labels on ghstack PRs.
+  # Note: as pull_request doesn't trigger on PRs targeting main,
+  # to test changes to the workflow itself one needs to create
+  # a PR that targets a gh/**/base branch.
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches: [gh/**/base]
```

Comment:
I think you may want to test is on regular PR (not created via ghstack) as well by removing this temporarily. We should see it run and catch.

Reply:
Wouldn't we want to keep `main` here as well since? Otherwise it will only run when using `ghstack`

Reply:
Nope, we need to remove main here because it's a security risk.  This workflow is granted write access, so it needs to get the `check_labels.py` script from main as part of `pull_request_target` trigger.  Basically, it will grant write access to random GitHub users (privileges escalation).  You see the ghstack part here because folks who are using ghstack already have write access to the repo.

Reply:
And don't worry about the check not running in your PR, you can land this and it will start running on all PRs after